File: app/src/main/java/com/example/biodun/mdb/Fragments/OverviewFragment.java
package com.example.biodun.mdb.Fragments;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.content.LocalBroadcastManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.biodun.mdb.Data.MovieData;
import com.example.biodun.mdb.DetailsPage;
import com.example.biodun.mdb.MainActivity;
import com.example.biodun.mdb.R;
import com.example.biodun.mdb.Utils.BuildUrl;
import com.example.biodun.mdb.Utils.ImageUtil;
import com.squareup.picasso.Picasso;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * A simple {@link Fragment} subclass.
 */
public class OverviewFragment extends Fragment {

    private MovieData movies =new MovieData() ;

    @BindView(R.id.overText) TextView overviewText;
    @BindView(R.id.name) TextView titleText;
    @BindView(R.id.date) TextView dateText;
    @BindView(R.id.textRating) TextView ratingText;
    @BindView(R.id.poster) ImageView image;

    public OverviewFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

           }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {


              if(this.getArguments()!=null){
              Bundle bundle=this.getArguments();
              movies=bundle.getParcelable("movies");
          }

        DetailsPage activity=(DetailsPage) getActivity();
        movies=activity.getMovies();
        String overview= movies.getOverview();
        String name=movies.getTitle();
        String date=movies.getReleaseDate();
        double rating=movies.getRatings();
        String url=movies.getImageUrl();
        String imageUrl= ImageUtil.getSmallImageUrl(url);
        String ratings=Double.toString(rating);

        View v= inflater.inflate(R.layout.fragment_overview, container, false);

            ButterKnife.bind(this,v);
        Picasso.with(getContext()).load(imageUrl).placeholder(R.drawable.poster1).into(image);
        overviewText.setText(overview);
        titleText.setText(name);
        dateText.setText(date);
        ratingText.setText(ratings);


        return v;
    }

    @Override
    public void onStart() {
        super.onStart();

    }



}

File: app/src/main/java/com/example/biodun/mdb/features/movieDetail/OverviewFragment.kt
package com.example.biodun.mdb.features.movieDetail

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import com.example.biodun.mdb.MdbApplication
import com.example.biodun.mdb.R
import com.example.biodun.mdb.Utils.ImageUtil
import com.example.biodun.mdb.di.ViewModelFactory
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.fragment_overview.*
import javax.inject.Inject

class OverviewFragment : Fragment() {

    @Inject
    lateinit var viewModelFactory: ViewModelFactory
    lateinit var viewModel: MovieDetailViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        return inflater.inflate(R.layout.fragment_overview, container, false)
    }

    override fun onResume() {
        super.onResume()
        displayMovieOverview()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        (context.applicationContext as MdbApplication).appComponent.inject(this)
        viewModel = ViewModelProviders.of(context as MovieDetailActivity, viewModelFactory)[MovieDetailViewModel::class.java]
    }

    private fun displayMovieOverview() {
        with(viewModel.getSelectedMovie()) {
            overText.text = overview
            overview_name.text = title
            overview_date.text = releaseDate
            textRating.text = ratings.toString()
            Picasso.with(context).load(ImageUtil.getSmallImageUrl(imageUrl))
                    .placeholder(R.drawable.poster1)
                    .into(poster)
        }
    }

    companion object {
        private var instance: OverviewFragment? = null

        fun getFragmentInstance(): OverviewFragment {
            if (instance == null) {
                synchronized(this) {
                    if (instance == null) {
                        instance = OverviewFragment()
                    }
                }
            }
            return instance!!
        }
    }
}
